src/fstar/FStar.Interactive.fs

```diff
@@ -67,41 +67,28 @@ type stack_t = list<(env_t * modul_t)>
 // Note: many of these functions are passing env around just for the sake of
 // providing a link to the solver (to avoid a cross-project dependency). They're
 // not actually doing anything useful with the environment you're passing it (e.g.
-// pop or reset_mark).
+// pop).
 
-let pop (_, env) msg =
-    pop_context env msg;
-    Options.pop()
+let push env msg =
+  let res = push_context env msg in
+  Options.push();
+  res
+
+let pop env msg =
+  pop_context (snd env) msg;
+  Options.pop()
 
 type push_kind = | SyntaxCheck | LaxCheck | FullCheck
 
-let push ((dsenv: DsEnv.env), tcenv) kind restore_cmd_line_options msg =
-    let tcenv = { tcenv with lax = (kind = LaxCheck) } in
-    let dsenv = DsEnv.set_syntax_only dsenv (kind = SyntaxCheck) in
-    let res = push_context (dsenv, tcenv) msg in
-    Options.push();
-    if restore_cmd_line_options then Options.restore_cmd_line_options false |> ignore;
-    res
-
-let mark (dsenv, env) =
-    let dsenv = DsEnv.mark dsenv in
-    let env = TcEnv.mark env in
-    Options.push();
-    dsenv, env
-
-let reset_mark (_, env) =
-    let dsenv = DsEnv.reset_mark () in
-    let env = TcEnv.reset_mark env in
-    Options.pop();
-    dsenv, env
+let push_with_kind ((dsenv: DsEnv.env), tcenv) kind restore_cmd_line_options msg =
+  let tcenv = { tcenv with lax = (kind = LaxCheck) } in
+  let dsenv = DsEnv.set_syntax_only dsenv (kind = SyntaxCheck) in
+  let res = push (dsenv, tcenv) msg in
+  if restore_cmd_line_options then Options.restore_cmd_line_options false |> ignore;
```

Comment:
No great reason; I just kept it consistent with the existing code.  This doesn't matter much, because these values are reset each time.  I'll fix it in a later commit :)